### DISCUSSION TITLE
Custom field type

### DISCUSSION THREAD

**Discussion by wei2go** [solspace/craft-freeform#283] [Questions & Answers] [answered]
I may have missed it in the doc, but I can't find a way to add a custom field type. Is there a way in Freeform to do so? Thanks.

**Comment by kjmartens** [comment, marked as answer]
Hi @wei2go, unfortunately there's currently no way to add a custom field type in Freeform yet. Out of curiosity, what kind of field type were you hoping to add? 🙂

**Reply by wei2go** [reply]
Hi @kjmartens ,

Thanks for the response.

I think Freeform already supports a very good set of fields which are sufficient most of time, but sometimes clients may have unique requirements, such as a custom signature field integrating with their Adobe Sign for example. So I wonder in that case if we have the flexibility to develop unique field types for particular clients.
